### PR TITLE
Additional programming environments for Kesch

### LINKED ISSUE #304: Fix programming environment settings for Kesch
We need to revert the changes for `PrgEnv-pgi` on Kesch and make available also alternative the `PrgEnv-pgi` modules. To summarise, the following must be changed:

1. `PrgEnv-pgi` environment on Kesch must use the default `PrgEnv-pgi` module.
2. Provide additional `PrgEnv-pgi` environments on Kesch (AJ's version, PGI 18.4)
3. Adapt `valid_prog_environs` on Kesch using wildcards.

### PR DESCRIPTION
**Pull request by ajocksch**
in execution of reframe

Closes #304

### CODE REVIEW

File: cscs-checks/prgenv/gpu_direct_acc.py [hunk 1]
```diff
@@ -29,14 +29,17 @@ def __init__(self, **kwargs):
...
+                                                       self.stdout)])
         self.maintainers = ['AJ', 'VK']
         self.tags = {'production'}
 
     def setup(self, partition, environ, **job_opts):
-        if environ.name == 'PrgEnv-cray':
+        if 'PrgEnv-cray' in environ.name:
             environ.fflags = '-hacc -hnoomp'
-        elif environ.name == 'PrgEnv-pgi':
+        elif 'PrgEnv-pgi' in environ.name:
```
Comment (vkarak) [inline cscs-checks/prgenv/gpu_direct_acc.py]: Perhaps, it's better to test if `environ.name.startswith('PrgEnv-pgi')`.

Comment (vkarak) [reply cscs-checks/prgenv/gpu_direct_acc.py]: Can you also move this file under `cscs-checks/mch/`?

File: cscs-checks/prgenv/gpu_direct_acc.py [hunk 2]
```diff
@@ -29,14 +29,17 @@ def __init__(self, **kwargs):
             self.num_tasks_per_node = 8
 
         self.sourcepath = 'gpu_direct_acc.F90'
-        self.sanity_patterns = sn.assert_found(r'Result :\s+OK', self.stdout)
+        self.sanity_patterns = sn.all([sn.assert_found(r'GPU with OpenACC',
+                                                       self.stdout),
+                                       sn.assert_found(r'Result :\s+OK',
+                                                       self.stdout)])
```
Comment (vkarak) [inline cscs-checks/prgenv/gpu_direct_acc.py]: Format this as follows; it's more readable:

```python
        self.sanity_patterns = sn.all([
            sn.assert_found(r'GPU with OpenACC', self.stdout),
            sn.assert_found(r'Result :\s+OK',self.stdout)
        ])
```

**Comment by teojgo** [comment]
Are these tests valid for dom/daint, because they also fail for me there.

File: cscs-checks/prgenv/gpu_direct_acc.py (hunk 2 above)
Comment (ajocksch) [reply cscs-checks/prgenv/gpu_direct_acc.py]: done

File: cscs-checks/prgenv/gpu_direct_acc.py (hunk 1 above)
Comment (ajocksch) [reply cscs-checks/prgenv/gpu_direct_acc.py]: done

**Comment by ajocksch** [comment]
added flag -Mnorpath ; now it is running on dom and daint

File: cscs-checks/prgenv/gpu_direct_acc.py [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
-from reframe.core.pipeline import RegressionTest
-
 
-class GpuDirectAccCheck(RegressionTest):
+@rfm.simple_test
+class GpuDirectAccCheck(rfm.RegressionTest):
     def __init__(self, **kwargs):
-        super().__init__('gpu_direct_acc_check',
-                         os.path.dirname(__file__), **kwargs)
+        super().__init__()
```
Comment (vkarak) [inline cscs-checks/prgenv/gpu_direct_acc.py]: 1. Are we happy with the auto-generated name?
2. Does it make sense to provide a short human-readable description of the test?

**Comment by vkarak** [comment]
@teojgo I think this is expected, isn't it @ajocksch?

**Comment by teojgo** [comment]
I tried myself on kesch and they fail

File: cscs-checks/prgenv/gpu_direct_acc.py (hunk 3 above)
Comment (vkarak) [reply cscs-checks/prgenv/gpu_direct_acc.py]: @ajocksch Can you please address/answer these? Other than that, this PR is ready to be merged.

File: cscs-checks/prgenv/gpu_direct_acc.py [hunk 4]
```diff
@@ -1,46 +1,52 @@
...
 
         super().setup(partition, environ, **job_opts)
-
-
-def _get_checks(**kwargs):
-    return [GpuDirectAccCheck(**kwargs)]
+        if (self.current_system.name in ['kesch']) and \
+            (environ.name.startswith('PrgEnv-pgi')):
+            self.job.launcher = LauncherWrapper(self.job.launcher,
+                'LD_PRELOAD=$X/libmpi.so', self.launch_options
```
Comment (vkarak) [inline cscs-checks/prgenv/gpu_direct_acc.py]: @ajocksch Why an `export LD_PRELOAD=...` wouldn't work? You could simply set `LD_PRELOAD` in `self.variables`.

**Pull request merged**